topology/host.go

```diff
@@ -63,6 +64,18 @@ func NewHostShardSet(host Host, shardSet sharding.ShardSet) HostShardSet {
 	return &hostShardSet{host, shardSet}
 }
 
+func newHostShardSetFromServiceInstance(si services.ServiceInstance, fn sharding.HashFn) (HostShardSet, error) {
+	shards := make([]uint32, 0, si.Shards().ShardsLen())
+	for _, s := range si.Shards().Shards() {
+		shards = append(shards, s.ID())
+	}
+	shardSet, err := sharding.NewShardSet(shards, fn)
+	if err != nil {
+		return nil, err
+	}
+	return &hostShardSet{host: NewHost(si.ID(), si.Endpoint()), shardSet: shardSet}, nil
```

Comment:
Any reason not to use the constructor here?  `NewHostShardSet(NewHost(si.ID(), si.Endpoint()), shardSet), nil`?